scripts/fires/wfigs_data.py

```diff
@@ -0,0 +1,227 @@
+"""Script to import and clean WFIGS data.
+"""
+from collections.abc import Sequence
+import datetime
+
+from absl import app
+from absl import logging
+import numpy as np
+import os
+import pandas as pd
+import pickle
+import re
+import requests
+import datacommons as dc
+import sys
+import mapping
+
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(_SCRIPT_PATH, '../../util/'))  # for recon util
+import latlng_recon_geojson
+
+pd.set_option("display.max_columns", None)
+
+PRE_2022_FIRE_LOCATIONS_URL = "https://services3.arcgis.com/T4QMspbfLg3qTGWY/arcgis/rest/services/Fire_History_Locations_Public/FeatureServer/0/query?where=1%3D1&outFields=InitialLatitude,InitialLongitude,InitialResponseAcres,InitialResponseDateTime,UniqueFireIdentifier,IncidentName,IncidentTypeCategory,IrwinID,FireCauseSpecific,FireCauseGeneral,FireCause,FireDiscoveryDateTime,ContainmentDateTime,ControlDateTime,IsCpxChild,CpxID,DiscoveryAcres,DailyAcres,POOFips,POOState,EstimatedCostToDate,TotalIncidentPersonnel,UniqueFireIdentifier&outSR=4326&f=json&resultType=standard"
+YTD_2022_FIRE_LOCATIONS_URL = "https://services3.arcgis.com/T4QMspbfLg3qTGWY/arcgis/rest/services/CY_WildlandFire_Locations_ToDate/FeatureServer/0/query?where=1%3D1&outFields=FireCauseGeneral,FireCauseSpecific,FireCause,InitialLatitude,InitialLongitude,InitialResponseAcres,InitialResponseDateTime,IrwinID,UniqueFireIdentifier,IncidentName,IncidentTypeCategory,FireDiscoveryDateTime,ContainmentDateTime,ControlDateTime,IsCpxChild,CpxID,DiscoveryAcres,DailyAcres,POOState,EstimatedCostToDate,TotalIncidentPersonnel,UniqueFireIdentifier&outSR=4326&f=json&resultType=standard"
+_OUTPUT = "/cns/jv-d/home/datcom/v3_resolved_mcf/fire/wfigs/"
+
+_FIRE_INCIDENT_MAP = {
+    'CX': 'FireIncidentComplexEvent',
+    'WF': 'WildlandFireEvent',
+    'RX': 'PrescribedFireEvent'
+}
+
+_LOCATION_MAP = "\nlocation: {location}"
+_DAILY_ACRES_MAP = "[Acre {daily_acres}]"
+_INITIAL_RESPONSE_AREA_MAP = "[Acre {initial_response_area}]"
+_EXPECTED_LOSS_MAP = "[USDollar {costs}]"
+
+
+def get_data(url):
+    """Get data from the API.
+
+  This method does that by retrieving the max results in a loop till we get to
+  the end of the dataset.
+
+  Args:
+    url: URL for the API call.
+
+  Returns:
+    A Pandas dataframe containing merged historical and YTD data.
+  """
+    r = requests.get(url)
+    x = r.json()
+    data = []
+    for row in x["features"]:
+        data.append(row["attributes"])
+    max_record_count = len(x["features"])
+    offset_str = "&resultOffset="
+    i = 1
+    while (x["features"]) and len(x["features"]) == max_record_count:
+        r = requests.get(url + offset_str + str(max_record_count * i))
```

Comment:
could you use a format string here to build the url? that will help with readability.

Reply:
Done